solutions/kattis-catandmice.mdx

```diff
@@ -0,0 +1,105 @@
+---
+id: kattis-catandmice
+source: Kattis
+title: Cat and Mice
+author: Kevin Sheng
+---
+
+## Explanation
+
+Binary search on the minimum possible speed of Cartesian Cat.
+
+For each speed, let $\texttt{mintime}[S][e]$ be the minimum time to go from the start to end $e$, eating each mouse in the set $S$ along the way.
+If it is impossible to do so, the value will be defined as infinity.
+When processing each state, go through all the possible previous ending points and see if it's possible to reach the defined end given the previous ending time. Calculate the minimum time for the state over all valid previous endings.
+
+## Implementation
+
+<LanguageSection>
+<JavaSection>
+```java
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.Arrays;
+
+public final class CatAndMice {
+	private static final long PRECISION = (long) 1e4;
+	public static void main(String[] args) throws IOException {
+		BufferedReader read = new BufferedReader(new InputStreamReader(System.in));
```

Comment:
use kattio

Reply:
wait what when was this required

Reply:
idk the format around here is to use Kattio so

Reply:
i don't think so? it's encourage but not required iirc

Reply:
I don't think it's required, but it might be nice to make the code more easy to understand. For example, this line
```mice[m] = Arrays.stream(read.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();```
is a bit unintuitive and I don't think many people understand what this method does.

Reply:
i've used it in solutions and it was ok'd by core before

Reply:
but harry always makes me use kattio though

Reply:
i mean using kattio doesn't do huge wonders for readability anyways